### ISSUE TITLE
hex programming

### ISSUE THREAD

**Issue by sg-** [xiongyihui/CMSIS-DAP#3]
I'm trying to work in your changes but have a problem with the following line.
https://github.com/xiongyihui/CMSIS-DAP/blob/arch_ble/interface/target/hal/DBG_NordicSemi/DBG_nRF51822/target_flash.c#L154

Given that flash.ram_to_flash_bytes_to_be_written is externally defined is doesnt seem safe to apply this logic, rather only write chunks of the size defined by the flash algorithm. Seems like a failure point but its late for me and hard to follow all the logic. Any ideas?

**Comment by xiongyihui** [comment]
[The flash algorithm](https://github.com/xiongyihui/CMSIS-DAP/blob/arch_ble/interface/flash_algo_mdk/nRF51822AA/FlashPrg.c#L193) supports writing different size chunks. For other targets, aligning  data chunks should be better.

**Comment by sg-** [comment]
Got it. So works for nordic but nothing else. I'll have a look at a more universal way to do this. Thanks.
